velox/exec/Spiller.cpp

```diff
@@ -565,26 +613,7 @@ void Spiller::fillSpillRuns(const RowContainerIterator* startRowIter) {
     for (;;) {
       auto numRows = container_->listRows(
           &iterator, rows.size(), RowContainer::kUnlimited, rows.data());
-      // Calculate hashes for this batch of spill candidates.
-      auto rowSet = folly::Range<char**>(rows.data(), numRows);
-
-      if (!isSinglePartition) {
-        for (auto i = 0; i < container_->keyTypes().size(); ++i) {
-          container_->hash(i, rowSet, i > 0, hashes.data());
-        }
-      }
-
-      // Put each in its run.
-      for (auto i = 0; i < numRows; ++i) {
-        // TODO: consider to cache the hash bits in row container so we only
-        // need to calculate them once.
-        const auto partition = isSinglePartition
-            ? 0
-            : bits_.partition(hashes[i], state_.maxPartitions());
-        VELOX_DCHECK_GE(partition, 0);
-        spillRuns_[partition].rows.push_back(rows[i]);
-        spillRuns_[partition].numBytes += container_->rowSize(rows[i]);
-      }
+      fillSpillRuns(rows, numRows, hashes, isSinglePartition);
```

Comment:
fillSpillRuns called here is more like calculate the partitions? Shall we call this function as partitionSpillRows instead? We don't need to pass isSinglePartition?

Reply:
I passed `hash` and `isSinglePartition` in this function as it could reuse the hash vector and only calculate once to get `isSinglePartition` outside the for loop.